sys/net/if_bridge.c

```diff
@@ -3470,6 +3470,7 @@ bridge_pfil(struct mbuf **mp, struct ifnet *bifp, struct ifnet *ifp, int dir)
 	}
 
 	/* Run the packet through pfil before stripping link headers */
+#ifdef INET
```

Comment:
Why is link-layer filtering conditional on INET? I'm not too familiar with this bit of the code, but it looks wrong.

Reply:
if we're an INET6-only kernel and the packet is an IPv4 packet, or vice versa, it will already be handled by the previous check on line 3448, which either drops the packet or unconditionally accepts it based on the value of `net.link.bridge.pfil_onlyip`, and we should never get here.

however, it does seem like there might be a pre-existing bug here when an INET6 kernel won't pass IPv6 packets to the filter.

i'm not sure under what circumstances `PFIL_HOOKED_OUT(V_link_pfil_head)` can be true while `PFIL_HOOKED_OUT(V_inet6_pfil_head)` is false -- perhaps this should be replaced by a check for `PFIL_HOOKED_OUT_ANY`?

Reply:
i tested this code:

```
        if (PFIL_HOOKED_OUT_ANY && V_pfil_ipfw != 0 && dir == PFIL_OUT && ifp != NULL) {
                switch (pfil_mbuf_out(V_link_pfil_head, mp, ifp, NULL)) {
                case PFIL_DROPPED:
                        return (EACCES);
                case PFIL_CONSUMED:
                        return (0);
                }
        }
```

and it does compile, but since i don't have any systems using bridge filtering, i can't really test it.

Reply:
I have no good way to test this at the moment either. I'd prefer to see this patch split into two: one which adds preprocessor symbols that let us avoid the ugly inline ifdefs, and one for the rest of the change. The former seems uncontroversial and a positive change, but I am not sure about the latter.